### ISSUE TITLE
Invalid URI should cause the promise to be rejected

### ISSUE THREAD

**Issue by josnidhin** [request/request-promise#43]
```js
var rq = require('request-promise');

rq('test').then(function(){
  console.log('success');
})
.catch(function(err){
  console.log('failed')
});
```

Expects the code to log 'failed' on console but instead throws error

**Comment by analog-nico** [comment]
Hi @josnidhin the culprit are [these lines](https://github.com/tyabonil/request-promise/blob/master/lib/rp.js#L82-L85). As you see there the behavior is intentional. It is replicating Request's original behavior in case no callback is registered. The reason is that the error you reported in particular is thrown before the `.then(...)` method is called. So at that point in time the code cannot differentiate between:
``` js
rq('test');
rq('test').then(...).catch(...);
```
By nature we have a situation here that allows no perfect solution in all cases. However, it could be the case that I can find a different solution that provides a better compromise. It will take me some time to work on, though.

**Comment by analog-nico** [comment]
I just released version 1.0.0 which fixes this issue.

Thanks for reporting this @josnidhin . Request-Promise embraces promises in a much more cleaner way.
